components/infra-proxy-service/server/proxy.go

```diff
@@ -109,3 +110,25 @@ func GetOrgAdminKeyFrom(secret *secrets.Secret) string {
 
 	return adminKey
 }
+
+// ParseAPIError parses common Chef Infra Server API errors into a user-readable format.
+func ParseAPIError(err error) error {
+	chefError, _ := chef.ChefError(err)
+	if chefError != nil {
+		switch chefError.StatusCode() {
+		case http.StatusBadRequest:
+			return status.Errorf(codes.InvalidArgument, chefError.StatusMsg())
```

Comment:
Use exact same message returned by Chef InfraServer API.

Reply:
Try to maintain the parity between HTTP error status code and GRPC error codes. ie.
`http.StatusBadRequest` might be equivalent to `codes.InvalidArgument`?